Remotion/Data/DomainObjects/ConfigurationLoader/ReflectionBasedConfigurationLoader/PropertyDefaultValueProvider.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq;
+using Remotion.Data.DomainObjects.Mapping;
+using Remotion.Reflection;
+using Remotion.ServiceLocation;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.ConfigurationLoader.ReflectionBasedConfigurationLoader
+{
+  /// <summary>
+  /// Implements the default behavior for <see cref="PropertyDefinition.DefaultValue"/>, which sets all reference-<see cref="Type"/> properties to
+  /// <see langword="null"/>, even if they are declared with <see cref="NullablePropertyAttribute.IsNullable"/> set to <see langword="false"/>.
+  /// </summary>     
+  [ImplementationFor(typeof(IPropertyDefaultValueProvider), Lifetime = LifetimeKind.Singleton, RegistrationType = RegistrationType.Single)]
+  public class PropertyDefaultValueProvider : IPropertyDefaultValueProvider
+  {
+    public object? GetDefaultValue (IPropertyInformation propertyInfo, bool isNullable)
+    {
+      if (isNullable)
+        return null;
+
+      var propertyType = propertyInfo.PropertyType;
+      if (propertyType.IsEnum)
+      {
+        var firstValueOrNull = EnumUtility.GetEnumMetadata(propertyType).OrderedValues.FirstOrDefault();
```

Comment:
Since this issue is only about reference types, the semantics for enum properties should not change. However, because the DefaultValue is now created when instantiating the PropertyDefinition instead of the first access to the DefaultValue property, it is important that we do not throw exceptions, so that the mapping validation can find the error. This is why even the LegacyPropertyDefaultValueProvider creates a zero value if no enum value is defined.

Reply:
I see. Follow up: do you think we should update the Enum in a separate ticket now that we are touching things here anyway and update the PropertyDefaultValueProvider to use the 0-value for Enum to better represent the .net default / unitialized backing field behavior?

Reply:
I lean towards yes we should do it, but no we should not create a different issue. This is because the behavior here is almost irrelevant because of the mapping validation. We can re-write the ticket to better reflect what we now know about the interaction between the PropertyDefaultValueProvider and the mapping validation.

Reply:
that's fine by me. I agree that the scope change is negligible in terms of the budget. we do should keep it as a separate commit though.

Reply:
For the record, we agreed upon not including this in the scope of this issue, because changing the behavior to initializing an invalid value would necessitate that we get rid of the value check when setting the property value and some other changes. This should be done in a separate issue.